lib/jsonapi/plugs/query_parser.ex

```diff
@@ -121,9 +125,14 @@ defmodule JSONAPI.QueryParser do
         |> Enum.into(MapSet.new())
 
       requested_fields =
-        value
-        |> String.split(",")
-        |> Enum.into(MapSet.new(), &String.to_atom/1)
+        try do
+          value
+          |> String.split(",")
+          |> Enum.map(&underscore/1)
```

Comment:
I think this could be considered a bug insofar that `UnderscoreParameters` doesn't catch this.